samcli/lib/providers/api_collector.py

```diff
@@ -145,12 +146,12 @@ def dedupe_function_routes(routes):
 
         Return
         -------
-        A list of routes without duplicate routes with the same function_name and method
+        A list of routes without duplicate routes with the same stack_path, function_name and method
         """
         grouped_routes = {}
 
         for route in routes:
-            key = "{}-{}".format(route.function_name, route.path)
+            key = "{}-{}-{}".format(route.stack_path, route.function_name, route.path)
```

Comment:
Will this mean people need to prepend stack name when calling a lambda locally?

Reply:
this key does not do anything customer facing. the key is used for dedup only.